sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryBaseTable.scala

```diff
@@ -962,6 +972,14 @@ private class BufferedRowsReader(
               wKeyType, wValueType)
           }
         }
+      case (readType: AtomicType, writeType: AtomicType) if readType != writeType =>
```

Comment:
it seems the InternalRow format requires cast of each element , if the nested type changes for collections like Map/Array.

Reply:
yes